[RT-703] Add documentation for container agent custom token secret (#7418)
circleci/circleci-docs: jekyll/_cci2/container-runner.adoc

          - name: "other"
```

[#custom-secret]

> === Custom Token Secret

Reviewer: ```suggestion
=== Custom token secret
```

> Using the above configuration will provision a Kubernetes secret containing your Resource Class tokens. In some circumstances you may wish to provision your own secret, or you simply might not want to specify the tokens via helm. Instead you can provision your own Kubernetes secret containing your tokens and specify its name in the `agent.customSecret` field.

Reviewer: ```suggestion
Using the configuration described above provisions a Kubernetes secret containing your resource class tokens. In some circumstances, you may wish to provision your own secret, or you simply might not want to specify the tokens via helm. Instead, you can provision your own Kubernetes secret containing your tokens and specify its name in the `agent.customSecret` field.
```

> The secret should contain a field for each Resource Class, using the Resource Class name as the key and the token as the value. Consider the following `resourceClasses` configuration:

Reviewer: ```suggestion
The secret should contain a field for each resource class, using the Resource Class name as the key and the token as the value. Consider the following `resourceClasses` configuration:
```

```yaml
agent:
  resourceClasses:  
    circleci-runner/resourceClass:
      metadata:
        annotations:
          custom.io: my-annotation
    

>     circleci-runner/resourceClass2:

Reviewer: For values that the reader needs to substitute for their own data, we use `<my-token>`, <my-resource-class> or whatever is appropriate. Could you update this snippet to use this format?

Reviewer: Oh and the one below to match

```

> The corresponding custom secret would have 2 fields:

Reviewer: ```suggestion

The corresponding custom secret would have 2 fields:
```

```yaml
circleci-runner.resourceClass: TOKEN1
circleci-runner.resourceClass2: TOKEN2
```

> Due to Kubernetes secret key character constraints, the `/` separating the Namespace and Resouce Class name is replaced with a `.` character. Other than this the name must exactly match the `resourceClasses` config to match the token with the correct configuration.

Reviewer: ```suggestion
Due to Kubernetes secret key character constraints, the `/` separating the namespace and resouce class name is replaced with a `.` character. Other than this, the name must exactly match the `resourceClasses` config to match the token with the correct configuration.
```

> Even if there is no further pod configuration, the resource class must be present in `resourceClasses` as an emtpy map, as shown by `circleci-runner/resourceClass2` in the above config

Reviewer: ```suggestion
Even if there is no further pod configuration, the resource class must be present in `resourceClasses` as an emtpy map, as shown by `circleci-runner/resourceClass2` in the above config example.
```

| `container-agent` (the name of the deployment)

| agent.resourceClasses *Default must be updated in order to run a job successfully*

> | Resource class task configuration. See "<<resource-class-configuration-custom-pod,Resource Class Configuration>>" section above
> | {}
> 
> | agent.customSecret
> | A user provided Kubernetes containing resource class tokens. See "<<custom-secret,Custom Token Secret>>" section above

Reviewer: ```suggestion
| Resource class task configuration. See the "<<resource-class-configuration-custom-pod,Resource Class Configuration>>" section above.
| {}

| agent.customSecret
| A user provided Kubernetes containing resource class tokens. See the "<<custom-secret,Custom Token Secret>>" section above.
```
